openml/runs/functions.py

```diff
@@ -762,6 +762,7 @@ def obtain_field(xml_obj, fieldname, from_server, cast=None):
     if "oml:output_data" not in run:
         if from_server:
             raise ValueError("Run does not contain output_data " "(OpenML server error?)")
+        predictions_url = None
```

Comment:
In the unit test this path was reached, and originally that would mean you got an `UnboundError` because `predictions_url` was never defined when creating the run.